src/main/java/com/fasterxml/jackson/databind/deser/JsonIgnoreValidator.java

```diff
@@ -0,0 +1,21 @@
+package com.fasterxml.jackson.databind.deser;
+
+import com.fasterxml.jackson.databind.annotation.JsonIgnoreIf;
+
+/**
+ * Abstract class which .class reference will be used to
+ * validate if a property annoated with {@link JsonIgnoreIf}
+ * should be ignored or not.
+ */
+public abstract class JsonIgnoreValidator {
+
+    /**
+     * When instantiating this class this abstract method should be defined.
+     * The logic will be executed and checked if property should be ignored
+     * or not.
+     *
+     * @return When returning true the property will not be parsed. When returning false not.
+     */
+    public abstract boolean ignore();
```

Comment:
Wouldn't it make sense to pass `AnnotatedMember` here?

Also, I think return type should be `Boolean` to allow distinguishing cases of "ignore", "do not ignore" and "no information". This may not be relevant for all use cases but in other cases last option is useful as it allows checking other annotations.